### PR TITLE
Fixed #2212 - Added 24 hour check to user settings

### LINKED ISSUE #2212: Display dates in localized formats
Right now, we're always spitting out "AM/PM" style dates.

This is a bit annoying for users who use 24 hour time, for example. AKA, anyone not in the US. And me personally since I use 24 hour time. :)

I did some quick Googling, but couldn't figure out a reliable way to detect this. For example, on my computer:

```javascript
> (new Date).toLocaleTimeString()
"3:49:17 PM"
```

/cc @mitsuhiko I feel like you might know this or have an opinion.

### PR DESCRIPTION
**Pull request by alexm92**
Also changed the `DateTime` and `TimeSince` to use user timezone and hour preferences.

### CODE REVIEW

**Comment by dcramer** [comment]
Thanks!

Backend looks good to me. @benvinegar can you take a look at the React bits?

File: src/sentry/static/sentry/app/components/dateTime.jsx [hunk 1]
```diff
@@ -7,14 +8,17 @@ const DateTime = React.createClass({
   },
 
   render() {
-    let date = this.props.date;
+    var date = this.props.date;
+    var user = ConfigStore.get('user');
+    var options = user ? user.options : {};
+    var format = options.clock24Hours ? 'MMMM D YYYY HH:mm:ss z' : 'LLL z';
```
Comment (benvinegar) [inline src/sentry/static/sentry/app/components/dateTime.jsx]: Please change to `let`

**Comment by alexm92** [comment]
Sure :)

**Comment by alexm92** [comment]
@benvinegar I did the change

**Comment by benvinegar** [comment]
:+1:

**Comment by dcramer** [comment]
Thanks!

**Comment by alexm92** [comment]
Awesome! Thanks for merging

**Comment by dcramer** [comment]
This is live on dotcom

**Comment by alexm92** [comment]
:+1:

**Pull request merged**